applications/asset_tracker_v2/src/cloud/azure_iot_hub_integration.c

```diff
@@ -9,6 +9,13 @@
 #include <logging/log.h>
 LOG_MODULE_REGISTER(MODULE, CONFIG_CLOUD_INTEGRATION_LOG_LEVEL);
 
+BUILD_ASSERT(IS_ENABLED(CONFIG_AGPS_SRC_SUPL) == 0,
```

Comment:
Could be a KConfig dependency instead.

Reply:
With the current implementation this would create a dependency in lib `azure_iot_hub` to lib `supl` . We currently switch in cloud integrations based on the Kconfig that enable the specific cloud libs. `CONFIG_AZURE_IOT_HUB` for instance, which reside in `subsys/net/lib/azure_iot_hub/kconfig`

We could look into creating app level kconfigs for switching in cloud integration layers, then we could add these Kconfig dependencies without messing with sub libraries.